codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/ServerCodegenVisitor.kt

```diff
@@ -637,6 +638,10 @@ open class ServerCodegenVisitor(
         rustCrate.withModule(ServerRustModule.Operation) {
             protocolGenerator.renderOperation(this, shape)
         }
+
+        // Postprocessing hook: generate structures not in the service closure.
```

Comment:
```suggestion
```

The hook allows for arbitrary structure shapes to be generated, they don't have to necessarily be outside of the service's closure. I've added a documentation suggestion to the decorator's method explaining why that would be bad.